compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/DocumentationLexer.java

```diff
@@ -669,23 +702,79 @@ private STToken readDocumentationBacktickContentToken() {
         if (nextToken == LexerTerminals.BACKTICK) {
             reader.advance();
             switchMode(ParserMode.DOCUMENTATION_INTERNAL);
+            resetSpecialKeywordInPlace();
             return getDocumentationSyntaxTokenWithNoTrivia(SyntaxKind.BACKTICK_TOKEN);
         }
 
+        int lookAheadCount = 0;
+        String backtickStr = "";
         while (!reader.isEOF()) {
             switch (nextToken) {
                 case LexerTerminals.BACKTICK:
                 case LexerTerminals.NEWLINE:
                 case LexerTerminals.CARRIAGE_RETURN:
                     break;
                 default:
-                    reader.advance();
-                    nextToken = peek();
+                    backtickStr = backtickStr.concat(String.valueOf((char) nextToken));
+                    lookAheadCount++;
+                    nextToken = reader.peek(lookAheadCount);
                     continue;
             }
             break;
         }
 
-        return getDocumentationLiteral(SyntaxKind.BACKTICK_CONTENT);
+        if (!isSpecialKeywordInPlace) {
+            // Look for a x(), m:x(), T.y(), m:T.y() match
+            boolean hasMatch = backtickStr.matches(backtickExprRegex);
```

Comment:
Shall we use our own method to visit the characters and do this validation? 
Regex-matching can be expensive. I think if we remove that regex, way we can also avoid Line #718 (string concat) which is also a bit expensive.
Usually numeric operations are much faster compared to string operations. eg: (int/char comparison vs string compare, array-access vs map-access)